### PR TITLE
Add role: hosted_engine_setup

### PR DESCRIPTION
**Pull request by arachmani**
Signed-off-by: Asaf Rachmani <arachman@redhat.com>

### CODE REVIEW

File: roles/hosted_engine_setup/README.md [hunk 1]
```diff
@@ -0,0 +1,374 @@
+# ovirt-ansible-hosted-engine-setup
+
+Ansible role for deploying oVirt Hosted-Engine
+
+# Requirements
+
+ * Ansible version 2.7 or higher
+ * Python SDK version 4.2 or higher
+ * Python netaddr library on the ansible controller node
+
```
Comment (mnecas) [inline roles/hosted_engine_setup/README.md:10]: Remove requirements if you need to add something to add it to main README.md

File: roles/hosted_engine_setup/README.md [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+* A fully qualified domain name prepared for your Engine and the host. Forward and reverse lookup records must both be set in the DNS.
+* `/var/tmp` has at least 5 GB of free space.
+* Unless you are using Gluster, you must have prepared storage for your Hosted-Engine environment (choose one):
+    * [Prepare NFS Storage](https://ovirt.org/documentation/admin-guide/chap-Storage/#preparing-nfs-storage)
+    * [Prepare ISCSI Storage](https://ovirt.org/documentation/admin-guide/chap-Storage/#preparing-iscsi-storage)
+* Install additional oVirt ansible role:
+
+    ```bash
+    $ ansible-galaxy install ovirt.engine-setup # case-sensitive
```
Comment (mnecas) [inline roles/hosted_engine_setup/README.md:25]: remove the installation... the role is installed automatically with collection

File: roles/hosted_engine_setup/README.md [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+## hosted_engine_deploy_localhost.yml
+
+```yml
+---
+- name: Deploy oVirt hosted engine
+  hosts: localhost
+  connection: local
+  roles:
+    - role: ovirt.hosted_engine_setup
+```
```
Comment (mnecas) [inline roles/hosted_engine_setup/README.md:113]: change all examples to work with the collection, not role by itself

File: roles/hosted_engine_setup/library/ovirt_disk_28.py [hunk 4]
```diff
@@ -0,0 +1,774 @@
+#!/usr/bin/python
```
Comment (mnecas) [inline roles/hosted_engine_setup/library/ovirt_disk_28.py:1]: remove the whole library it is not necessary use the modules from collection

File: roles/hosted_engine_setup/tasks/create_target_vm/01_create_target_hosted_engine_vm.yml [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+        '[','').replace(']','').split(', ')|first
+  - name: Get storage domain details
+    ovirt_storage_domain_info:
+      pattern: name={{ he_storage_domain_name }} and datacenter={{ datacenter_name }}
+      auth: "{{ ovirt_auth }}"
+    register: storage_domain_details
+  - debug: var=storage_domain_details
+  - name: Add HE disks
+    # rename it to ovirt_disk with ansible 2.8+
+    ovirt_disk_28:
```
Comment (mnecas) [inline roles/hosted_engine_setup/tasks/create_target_vm/01_create_target_hosted_engine_vm.yml:126]: change to `ovirt_disk`

**Review by mnecas — CHANGES_REQUESTED** [review]
Hi,
please read the wiki around migrating the roles
https://github.com/oVirt/ovirt-ansible-collection/wiki/Development#migrating-and-adding-roles 
and also check out the CI of the bot so you pass all tests

File: changelogs/fragments/add-hosted_engine_setup-role.yml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+---
+    major_changes:
```
Comment (mnecas) [inline changelogs/fragments/add-hosted_engine_setup-role.yml:2]: please remove the extra tab so we have it align with other fragments

File: changelogs/fragments/add-hosted_engine_setup-role.yml [hunk 7]
```diff
@@ -0,0 +1,3 @@
...
+      - hosted_engine_setup - Migrate role (https://github.com/oVirt/ovirt-ansible-collection/pull/106).
```
Comment (mnecas) [inline changelogs/fragments/add-hosted_engine_setup-role.yml:3]: at the end add extra line

File: README.md [hunk 8]
```diff
@@ -24,6 +24,7 @@ Requirements
 
  * Ansible version 2.9 or higher
  * Python SDK version 4.4 or higher
+ * Python netaddr library on the ansible controller node
```
Comment (mnecas) [inline README.md:27]: add the library to spec.in file and to bindep.txt/requirements.txt

File: roles/hosted_engine_setup/examples/nfs_deployment.json [hunk 9]
```diff
@@ -0,0 +1,8 @@
+{
+    "he_bridge_if": "eth0",
+    "he_fqdn": "he-engine.example.com",
+    "he_vm_mac_addr": "00:a5:3f:66:ba:12",
+    "he_domain_type": "nfs",
+    "he_storage_domain_addr": "192.168.100.50",
+    "he_storage_domain_path": "/var/nfs_folder"
+}
```
Comment (mnecas) [inline roles/hosted_engine_setup/examples/nfs_deployment.json:8]: also, add a blank line at the end

**Review by mnecas — CHANGES_REQUESTED** [review]
Just a few small nits.
The CI fails on a few `208 File permissions not mentioned`.
Great job!

File: ovirt-ansible-collection.spec.in [hunk 10]
```diff
@@ -12,8 +12,11 @@ Group:          Virtualization/Management
 BuildArch:      noarch
 Url:            http://www.ovirt.org
 
-Requires: ansible >= 2.9.0
+Requires: ansible >= 2.9.11
```
Comment (mnecas) [inline ovirt-ansible-collection.spec.in:15]: What is the reason for this change?
If it is required by hosted engine update also readme that we require ansible "2.9.11 or higher"

Comment (arachmani) [reply ovirt-ansible-collection.spec.in:15]: Ansible changed the behavior in 2.9.11 and HE deployment failed (https://bugzilla.redhat.com/show_bug.cgi?id=1859483)
I'll update the README.

**Comment by arachmani** [comment]
Verified

**Review by mnecas — APPROVED** [review]
LGTM
